From c049d48407ff0459ff15e466edeeee6ecff67fcd Mon Sep 17 00:00:00 2001
From: Jaroslav Kysela <perex@perex.cz>
Date: Fri, 1 Oct 2010 14:08:03 +0200
Subject: [PATCH] namehint: Another fix to properly evaluate hw devices

Signed-off-by: Jaroslav Kysela <perex@perex.cz>
---
 src/control/namehint.c |    5 +++--
 1 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/src/control/namehint.c b/src/control/namehint.c
index ab3525e..faaa5d5 100644
--- a/src/control/namehint.c
+++ b/src/control/namehint.c
@@ -439,11 +439,12 @@
 			err = next_devices[list->iface](list->ctl, &device);
 			if (device < 0)
 				err = -EINVAL;
+			else
+				max_device = device;
 			while (err >= 0 && device >= 0) {
 				err = next_devices[list->iface](list->ctl, &device);
-				if (device > max_device)
+				if (err >= 0 && device > max_device)
 					max_device = device;
-				ok++;
 			}
 			ok = 0;
 			for (device = 0; err >= 0 && device <= max_device; device++) {
-- 
1.7.6

