From 7a2cdb33c4dba2258e074c80ea0227629113e376 Mon Sep 17 00:00:00 2001
From: Jaroslav Kysela <perex@perex.cz>
Date: Tue, 13 Apr 2010 10:08:56 +0200
Subject: [PATCH] pcm_share plugin: fix pcm->monotonic setup in open()
 function

Signed-off-by: Jaroslav Kysela <perex@perex.cz>
---
 src/pcm/pcm_share.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/pcm/pcm_share.c b/src/pcm/pcm_share.c
index c8a2d20..56a8685 100644
--- a/src/pcm/pcm_share.c
+++ b/src/pcm/pcm_share.c
@@ -1529,7 +1529,7 @@ int snd_pcm_share_open(snd_pcm_t **pcmp, const char *name, const char *sname,
 	pcm->private_data = share;
 	pcm->poll_fd = share->client_socket;
 	pcm->poll_events = stream == SND_PCM_STREAM_PLAYBACK ? POLLOUT : POLLIN;
-	pcm->monotonic = pcm->monotonic;
+	pcm->monotonic = slave->pcm->monotonic;
 	snd_pcm_set_hw_ptr(pcm, &share->hw_ptr, -1, 0);
 	snd_pcm_set_appl_ptr(pcm, &share->appl_ptr, -1, 0);
 
-- 
1.7.6

